lib/compilers/glimmer/index.js

```diff
@@ -12,13 +12,28 @@ export default class GlimmerTemplatePrecompiler extends Filter {
   }
 
   processString(content, relativePath) {
-    let specifier = getTemplateSpecifier(this.rootName, relativePath);
-    return `export default ${this.precompile(content, { meta: { specifier, '<template-meta>': true } })};`;
+    let specifier = getTemplateSpecifier(this.rootName, removeManagerIdFromPath(relativePath));
+    let managerId = getManagerIdFromPath(relativePath);
+    let meta = { specifier };
+
+    if (managerId) {
+      meta.managerId = managerId;
+    }
+
+    return `export default ${this.precompile(content, { meta })};`;
   }
 
   precompile(content, options) {
     return precompile(content, options);
   }
+
+  getDestFilePath(relativePath) {
+    let processedPath = super.getDestFilePath(relativePath);
+
+    if (!processedPath) return processedPath;
+
+    return removeManagerIdFromPath(processedPath);
```

Comment:
Indeed, resolution-map-builder does need to be fixed as well. I'm working on another PR that takes care of this. Basically—if I'm correct that we can't import a module named `'some/path/template.fragment'`—then we need to somewhere in the build process strip the moduleId from the filename (which I'm doing in this PR) _and_ modify resolution-map-builder to strip moduleIds. We have to do it twice because constructing the module map happens before template compilation (and needs to because we need the .hbs extensions for module map construction afaik).